web/src/client/network.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import { applyMixin, withDBus } from "./mixins";
+
+const NM_IFACE = "org.freedesktop.NetworkManager";
+
+export default class NetworkClient {
+  constructor(dbusClient) {
+    this._client = dbusClient;
+  }
+
+  async config() {
+    const data = await this.addresses();
+    const addrs = data.map(a => {
+      return {
+        address: a.address.v,
+        prefix: a.prefix.v
+      };
+    });
+
+    return {
+      addresses: addrs,
+      hostname: await this.hostname()
+    };
+  }
+
+  async connections() {
```

Comment:
We know that the NM API is complex.
Please add links to NM reference documentation so that in a few weeks/months we can understand what's going on.
Like this for example:

```suggestion
/**
 * Return object paths implementing o.fd.NM.Connection.Active
 *  https://developer-old.gnome.org/NetworkManager/stable/gdbus-org.freedesktop.NetworkManager.Connection.Active.html
 */
  async connections() {
```

Reply:
BTW if you care about the rendered APi docs (optional), this should work, while #155 is not done:

```console
cd web
npx jsdoc src
xdg-open out/index.html
```